src/main/scala/ai/starlake/schema/generator/DagGenerateCommand.scala

```diff
@@ -110,31 +110,40 @@ class DagGenerateCommand(schemaHandler: SchemaHandler) extends LazyLogging {
       DagPair(k, v)
     }.toList
     val depsEngine = new AutoTaskDependencies(settings, schemaHandler, settings.storageHandler())
-    taskConfigs.foreach { taskConfig =>
-      val dagTemplateName = taskConfig.dagConfig.template
-      val dagTemplateContent = Yml2DagTemplateLoader.loadTemplate(dagTemplateName)
-      val cron = settings.appConfig.schedulePresets.getOrElse(
-        taskConfig.schedule.getOrElse("None"),
-        taskConfig.schedule.getOrElse("None")
-      )
-      val cronIfNone = if (cron == "None") null else cron
-      val config = AutoTaskDependenciesConfig(tasks = Some(List(taskConfig.taskDesc.name)))
-      val deps = depsEngine.jobsDependencyTree(config)
-      val filename = Utils.parseJinja(
-        taskConfig.dagConfig.filename,
-        schemaHandler.activeEnvVars() ++ Map(
-          "table"  -> taskConfig.taskDesc.table,
-          "domain" -> taskConfig.taskDesc.domain,
-          "name"   -> taskConfig.taskDesc.name
+    val taskConfigsGroupedByFilename = taskConfigs
+      .map { taskConfig =>
+        val filename = Utils.parseJinja(
+          taskConfig.dagConfig.filename,
+          schemaHandler.activeEnvVars() ++ Map(
+            "table"  -> taskConfig.taskDesc.table,
+            "domain" -> taskConfig.taskDesc.domain,
+            "name"   -> taskConfig.taskDesc.name
+          )
         )
-      )
-      val context = TransformDagGenerationContext(
-        config = taskConfig.dagConfig,
-        deps = deps,
-        cron = Option(cronIfNone)
-      )
-      applyJ2AndSave(outputDir, jEnv, dagTemplateContent, context.asMap, filename)
-    }
+        (filename, taskConfig)
+      }
+      .groupBy { case (filename, _) => filename }
+    taskConfigsGroupedByFilename
+      .foreach { case (filename, taskConfigs) =>
+        val headConfig = taskConfigs.head match { case (_, config) => config }
+        val dagConfig = headConfig.dagConfig
+        val dagTemplateName = dagConfig.template
```

Comment:
We can add this test but I am not sure that it is relevant.

Indeed if the tasks use different dag templates, which may absolutely be the case, it means that in this particular case they will end with dags overwriting the previous ones. 

In other words, it would mean that those different configurations would use exactly the same filename which should not be the case.

Reply:
Yes, this can happen when user misconfigured dag generation. It allows the user to easily identify this mistake and modify dag generation config accordingly